enumerable.rb

```diff
@@ -1,50 +1,88 @@
 module Enumerable
-    def my_each
-        if block_given?
-            for x in self do
-                yield (x)
-            end
-        end
-        return self
-    end
-    
-    def my_each_with_index
-        if block_given?
-            for x in self do
-                yield (x)                
-            end
-            for y in self do
-                y.index(self[x]).to_i
-            end
-        end
-        return self
-    end
-    ['a', 'b', 'c'].my_each_with_index { |el, i| puts i }
-    def my_select
-
+  def my_each
+    if block_given?
+      each do |x|
+        yield x
+      end
     end
+    self
+  end
 
-    def my_all?
-
+  def my_each_with_index
+    if block_given?
+      idx = 0
+      each do |x|
+        yield(x, idx)
+        idx += 1
+      end
     end
+    self
+  end
 
-    def my_any?
+  def my_select
+    return unless block_given?
 
+    array = []
+    each do |x|
+      array.push(self[x]) if yield(self[x]) == true
     end
+    array
+  end
 
-    def my_none?
+  def my_all?
+    my_each { |x| return false if yield(x) == false }
+    true
+  end
 
-    end
+  def my_any?
+    my_each { |x| return true if yield(x) == true }
+    false
+  end
 
-    def my_count
+  def my_none?
+    my_each { |x| return false if yield(x) == true }
+    true
+  end
 
+  def my_count(par = nil)
+    number = 0
+    if block_given?
+      my_each do |x|
+        number += 1 if yield(x) == true
+      end
+    elsif !par.nil?
+      my_each do |x|
+        number += 1 if x == par
+      end
+    else
+      number = length
     end
+    number
+  end
 
-    def my_map
-
+  def my_map()
+    arr = []
+    if block_given?
+      my_each do |x|
+        arr.push(yield(x))
+      end
     end
+    arr
+  end
 
-    def my_inject
+  def my_inject(par = nil)
+    if par.nil?
+      acc = self[0]
+      (1...length).each do |x|
+        acc = yield(acc, self[x])
+      end
```

Comment:
#### my_inject 

- [ ] combines all elements of enum by applying a binary operation, specified by a block

Given
```
    operation = proc { |sum, n| sum + n } 
```
Expected result

    > (1..5).my_inject(4) { |prod, n| prod * n } should be equal (1..5).inject(4) { |prod, n| prod * n } => 480

- [ ] When a symbol is provided as an argument, my_inject should combine the elements of the enum using that symbol as a named method.

     > this [2, 5, 6].my_inject(:+) should be equal to this [2, 5, 6].inject(:+) => 13